### PR TITLE
fix: shrink SelectWidget label only if value is not empty. fixes #3369

### LINKED ISSUE #3369: Material UI Select widget label always shrinked
The `SelectWidget` always uses a shrinked label:
https://github.com/rjsf-team/react-jsonschema-form/blob/v5.0.0-beta.16/packages/material-ui/src/SelectWidget/SelectWidget.tsx#L67

The `shrink` property should be dependent on the current value, whether it is empty or not

I can provide a PR if it make sense.

### PR DESCRIPTION
**Pull request by ValYouW**
Fix https://github.com/rjsf-team/react-jsonschema-form/issues/3369

### CODE REVIEW

File: packages/material-ui/src/SelectWidget/SelectWidget.tsx [hunk 1]
```diff
@@ -74,11 +80,11 @@ export default function SelectWidget<
       select // Apply this and the following props after the potential overrides defined in textFieldProps
       InputLabelProps={{
         ...textFieldProps.InputLabelProps,
-        shrink: true,
+        shrink: !_isEmpty(),
       }}
       SelectProps={{
         ...textFieldProps.SelectProps,
-        multiple: typeof multiple === "undefined" ? false : multiple,
+        multiple: multiple,
```
Comment (heath-freenome) [inline packages/material-ui/src/SelectWidget/SelectWidget.tsx:87]: You can use the shortcut
```suggestion 
        multiple,
```

File: packages/mui/src/SelectWidget/SelectWidget.tsx [hunk 2]
```diff
@@ -74,11 +80,11 @@ export default function SelectWidget<
       select // Apply this and the following props after the potential overrides defined in textFieldProps
       InputLabelProps={{
         ...textFieldProps.InputLabelProps,
-        shrink: true,
+        shrink: !_isEmpty(),
       }}
       SelectProps={{
         ...textFieldProps.SelectProps,
-        multiple: typeof multiple === "undefined" ? false : multiple,
+        multiple: multiple,
```
Comment (heath-freenome) [inline packages/mui/src/SelectWidget/SelectWidget.tsx:87]: You can use the shortcut
```suggestion
        multiple,
```

File: packages/material-ui/src/SelectWidget/SelectWidget.tsx [hunk 3]
```diff
@@ -43,7 +43,13 @@ export default function SelectWidget<
 }: WidgetProps<T, S, F>) {
   const { enumOptions, enumDisabled } = options;
 
+  multiple = typeof multiple === "undefined" ? false : !!multiple;
+
   const emptyValue = multiple ? [] : "";
+  const _isEmpty = () =>
+    typeof value === "undefined" ||
+    (multiple && value.length < 1) ||
+    (!multiple && value === emptyValue);
```
Comment (heath-freenome) [inline packages/material-ui/src/SelectWidget/SelectWidget.tsx:52]: Why make this a function? You could simply assign it to a local variable and pass that instead?

File: packages/mui/src/SelectWidget/SelectWidget.tsx [hunk 4]
```diff
@@ -43,7 +43,13 @@ export default function SelectWidget<
 }: WidgetProps<T, S, F>) {
   const { enumOptions, enumDisabled } = options;
 
+  multiple = typeof multiple === "undefined" ? false : !!multiple;
+
   const emptyValue = multiple ? [] : "";
+  const _isEmpty = () =>
+    typeof value === "undefined" ||
+    (multiple && value.length < 1) ||
+    (!multiple && value === emptyValue);
```
Comment (heath-freenome) [inline packages/mui/src/SelectWidget/SelectWidget.tsx:52]: Why make this a function? You could simply assign it to a local variable and pass that instead?

**Review by heath-freenome — COMMENTED** [review]
Also, please update the `CHANGELOG.md` file to update the `beta-18` section for `@rjsf/material-ui` and `@rjsf/mui`

File: packages/material-ui/src/SelectWidget/SelectWidget.tsx (hunk 3 above)
Comment (ValYouW) [reply packages/material-ui/src/SelectWidget/SelectWidget.tsx]: Old class components habits... fixed.

**Pull request merged**